modules/oe_content_event/tests/src/Kernel/EventLinkSourceFilterTest.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_content_event\Kernel;
+
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\node\Entity\Node;
+use Drupal\oe_content_event\Plugin\InternalLinkSourceFilter\EventPeriodFilter;
+
+/**
+ * Tests the internal link source filters related to Events.
+ *
+ * @covers \Drupal\oe_content_event\Plugin\InternalLinkSourceFilter\EventPeriodFilter
+ */
+class EventLinkSourceFilterTest extends EventKernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'oe_link_lists',
+    'oe_link_lists_internal_source',
+  ];
+
+  /**
+   * Test that the event filter works as intended.
+   */
+  public function testEventLinkSourceFilter(): void {
```

Comment:
Let's add https://www.drupal.org/project/datetime_testing and use this for testing properly the times. We are already using in the theme for the Behat extension.